From b0d66e7b0980bde35bc7276e9bf40727d426ff31 Mon Sep 17 00:00:00 2001
From: Andrew Cooper <andrew.cooper3@citrix.com>
Date: Sun, 6 Jul 2014 20:02:39 +0100
Subject: [PATCH 1/1] tools/python: Improve .gitgnore and clean Makefile rule

Ingore all intermediate python files, and use find in the clean rule as there
are no Makefiles in subdirectories to participate in a recursive clean.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
CC: Ian Campbell <Ian.Campbell@citrix.com>
CC: Ian Jackson <Ian.Jackson@eu.citrix.com>
---
 .gitignore            |    1 +
 tools/python/Makefile |    8 +++-----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff -r cb5f649f32c8 .gitignore
--- a/.gitignore
+++ b/.gitignore
@@ -16,6 +16,7 @@
 *.spit
 *.gcno
 *.gcda
+*.py[ocd]
 TAGS
 GTAGS
 GRTAGS
diff -r cb5f649f32c8 tools/python/Makefile
--- a/tools/python/Makefile
+++ b/tools/python/Makefile
@@ -19,23 +19,9 @@ build: genpath genwrap.py $(XEN_ROOT)/to
 	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py build --xend=$(CONFIG_XEND)
 
 .PHONY: install
-install: install-dtd
-	CC="$(CC)" CFLAGS="$(CFLAGS)" $(PYTHON) setup.py install \
-		$(PYTHON_PREFIX_ARG) --root="$(DESTDIR)" --force --xend=$(CONFIG_XEND)
-
-	$(INSTALL_DIR) $(DESTDIR)$(SBINDIR)
-ifeq ($(CONFIG_XEND),y)
-	$(INSTALL_PYTHON_PROG) xen/xm/xm $(DESTDIR)$(SBINDIR)/xm
-	$(INSTALL_PYTHON_PROG) xen/xend/xend $(DESTDIR)$(SBINDIR)/xend
-endif
-
-install-dtd: all
-ifeq ($(CONFIG_XEND),y)
-	$(INSTALL_DIR) $(DESTDIR)$(SHAREDIR)/xen
-	$(INSTALL_DATA) xen/xm/create.dtd $(DESTDIR)$(SHAREDIR)/xen
-else
-	:
-endif
+install:
+	CC="$(CC)" CFLAGS="$(CFLAGS) $(APPEND_LDFLAGS)" $(PYTHON) setup.py install \
+		$(PYTHON_PREFIX_ARG) --root="$(DESTDIR)" --force
 
 .PHONY: test
 test:
@@ -43,10 +29,8 @@ test:
 
 .PHONY: clean
 clean:
-	rm -f $(XENPATH)
-	rm -rf build/ *.pyc *.pyo *.o *.a *~ xen/util/auxbin.pyc
-	rm -f xen/lowlevel/xl/_pyxl_types.h
-	rm -f xen/lowlevel/xl/_pyxl_types.c
-	rm -f $(DEPS)
+	find . \( -name "*.py[ocd]" -o -name "*~" -o -name "_*.[hc]" \) -delete
+	rm -rf build/
+	rm -f $(XENPATH) $(DEPS)
 
 -include $(DEPS)
diff -r cb5f649f32c8 tools/python/setup.py
--- a/tools/python/setup.py
+++ b/tools/python/setup.py
@@ -89,7 +89,7 @@ xl = Extension("xl",
 
 modules = [ xc, xs, ptsname ]
 
-enable_xend = True
+enable_xend = False
 new_argv = []
 for arg in sys.argv:
     if arg == "--xend=y":
